server/filestore.go

```diff
@@ -546,8 +547,12 @@ const (
 // This is the max room needed for index header.
 const indexHdrSize = 7*binary.MaxVarintLen64 + hdrLen + checksumSize
 
+// Lock held on entry
 func (fs *fileStore) recoverMsgBlock(fi os.FileInfo, index uint64) (*msgBlock, error) {
 	mb := &msgBlock{fs: fs, index: index, cexp: fs.fcfg.CacheExpire}
+	if len(fs.cfg.Subjects) == 1 {
```

Comment:
So on further reflection, this might need to be double checked. For instance if I start a stream with ["foo", "bar"] and later update to just ["foo"] this might not be correct.

We could use this as a hint, but for this code once we read it back in and build the fss state we know the real answer.

So after rebuilding fss for this block, we can do..

```
if len(mb.fss) == 1 {
  for sfilter := range mb.fss {
    mb.sfilter = sfilter 
  }
}

Reply:
Actually since we always build fss we should only do this logic when recovering a block. That way if the stream has multiple subjects but the block only has 1 we can optimize.